aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/worker/BaseTransferWorker.kt

```diff
@@ -150,8 +150,8 @@ internal abstract class BaseTransferWorker(
         )
     }
 
-    private fun isRetryableError(e: Throwable?): Boolean {
-        return isStopped ||
+    private suspend fun isRetryableError(e: Throwable?): Boolean {
```

Comment:
You can move the `isActive` check inline in `doWork` - this function should not need to suspend.